R/generateFilterValues.R

```diff
@@ -125,21 +125,26 @@ generateFilterValuesData = function(task, method = "randomForestSRC_importance",
     }
 
   } else {
-    fval = lapply(filter, function(x) {
-      x = do.call(x$fun, c(list(task = task, nselect = nselect), more.args[[x$name]]))
-      missing.score = setdiff(fn, names(x))
-      x[missing.score] = NA_real_
-      x[match(fn, names(x))]
-    })
+    index_names = names(method)
+	if (is.null(index_names))
+		index_names = method
```

Comment:
Do we want to allow this in the first place? Or assert this and stop early?

Reply:
This is for backwards compatibility. You can still pass a vector of filters in argument method: 
```
#' @examples
#' # two simple filter methods
#' fval = generateFilterValuesData(iris.task,
#'   method = c("FSelectorRcpp_gain.ratio", "FSelectorRcpp_information.gain"))
#' # using ensemble method "E-mean"
#' fval = generateFilterValuesData(iris.task,
#'   method = list("E-mean", c("FSelectorRcpp_gain.ratio", "FSelectorRcpp_information.gain")))
```
In this case names(method) will be NULL and so we want the column names of fval (prior to melting) to be the names of the filters that were passed in.

Reply:
Ah I see, and because we are now allowing user-defined names via named lists, we need to check if `index_names` are NULL in fact.

(PS: Please try to format your code blocks as R code and also use inline-code highlighting. This makes comments easier to read. Thanks.)